src/main/java/com/hyperwallet/clientsdk/util/MultipartRequest.java

```diff
@@ -43,11 +39,23 @@ public void setMultipartList(Multipart multipartList) {
         this.multipartList = multipartList;
     }
 
-    public Response putResource(boolean usesProxy, Proxy proxy) throws IOException {
+    public Response putResource(boolean usesProxy, Proxy proxy, final String proxyUsername, final String proxyPassword) throws IOException {
```

Comment:
Minor: Maybe we can split this into two methods, one that uses proxy and one that doesn't?  Then in HyperwalletApiClient you'd have:
```
response = usesProxy() ? getMultipartService(url, uploadData).putResource(getProxy(), getProxyUsername(), getProxyPassword()) : getMultipartService(url, uploadData).putResource();
```

Reply:
Hmm Idk if I agree, since the only real code diff is three lines

```
        if (proxyUsername != null && proxyPassword != null) {
        // NOTE: Removing Basic Auth from tunneling disabledSchemas is required in order
        // for Proxy Authorization to work. To prevent overriding client System Settings,
        // the client should set this System property themselves inside their JVM Options (1)
        // or their own code (2). Approaches listed below:
        // 1. jdk.http.auth.tunneling.disabledSchemes=
        // 2. System.setProperty("jdk.http.auth.tunneling.disabledSchemes", "false");
            Authenticator authenticator = new DefaultPasswordAuthenticator(
                    proxyUsername, proxyPassword);
            Authenticator.setDefault(authenticator);
        }
```